src/test/unit_tests/test_bucketspace_fs.js

```diff
@@ -954,6 +955,65 @@ mocha.describe('bucketspace_fs', function() {
+        });
+        mocha.it('get bucket policy not allowed non owner user', async function() {
+            const policy = generate_s3_policy(account_user1.name, test_bucket3, ['s3:*']).policy;
+            const param = { name: test_bucket3, policy: policy };
+            await bucketspace_fs.put_bucket_policy(param, dummy_object_sdk);
+            const res = await bucketspace_fs.get_bucket_policy(param, dummy_object_sdk);
+            assert_bucket_policies(policy, res.policy);
+            const dummy_object_sdk_not_owner = make_dummy_object_sdk_for_account(dummy_object_sdk, account_user1);
+            try {
+                await bucketspace_fs.get_bucket_policy(param, dummy_object_sdk_not_owner);
+                assert.fail('should have failed with METHOD_NOT_ALLOWED');
+            } catch (err) {
+                assert.ok(err.rpc_code === 'METHOD_NOT_ALLOWED');
+            }
+        });
+
+        mocha.it('put bucket policy not allowed non owner user', async function() {
+            const policy = generate_s3_policy(account_user1.name, test_bucket3, ['s3:*']).policy;
+            const param = { name: test_bucket3, policy: policy };
+            await bucketspace_fs.put_bucket_policy(param, dummy_object_sdk);
+            const res = await bucketspace_fs.get_bucket_policy(param, dummy_object_sdk);
+            assert_bucket_policies(policy, res.policy);
+            const dummy_object_sdk_not_owner = make_dummy_object_sdk_for_account(dummy_object_sdk, account_user1);
+            try {
+                await bucketspace_fs.put_bucket_policy(param, dummy_object_sdk_not_owner);
+                assert.fail('should have failed with METHOD_NOT_ALLOWED');
+            } catch (err) {
+                assert.ok(err.rpc_code === 'METHOD_NOT_ALLOWED');
+            }
+        });
+
+        mocha.it('delete bucket policy not allowed non owner user', async function() {
+            const policy = generate_s3_policy(account_user1.name, test_bucket3, ['s3:*']).policy;
+            const param = { name: test_bucket3, policy: policy };
+            await bucketspace_fs.put_bucket_policy(param, dummy_object_sdk);
+            const res = await bucketspace_fs.get_bucket_policy(param, dummy_object_sdk);
+            assert_bucket_policies(policy, res.policy);
+            const dummy_object_sdk_not_owner = make_dummy_object_sdk_for_account(dummy_object_sdk, account_user1);
+            try {
+                await bucketspace_fs.delete_bucket_policy(param, dummy_object_sdk_not_owner);
+                assert.fail('should have failed with METHOD_NOT_ALLOWED');
+            } catch (err) {
+                assert.ok(err.rpc_code === 'METHOD_NOT_ALLOWED');
+            }
+            await bucketspace_fs.delete_bucket_policy(param, dummy_object_sdk);
+            const resp = await bucketspace_fs.get_bucket_policy(param, dummy_object_sdk);
+            assert_bucket_policies(undefined, resp.policy);
```

Comment:
I would add a comment that this step is for cleaning the bucket policy.

Reply:
its more like testing the delete bucket policy working for owner or not, not exactly a clean up

Reply:
Good, thanks.
I think it can be helpful if you would add a comment that this part was to test that the deletion of the bucket policy works for the bucket owner (in the test title "delete bucket policy not allowed non owner user").